tmt/result.py

```diff
@@ -54,7 +54,18 @@ def is_result_outcome(cls, value: 'ResultInterpret') -> bool:
     }
 
 
-ResultData = Dict[str, Any]
+@dataclasses.dataclass
+class ResultData():
+    """
+    Formal data class containing information about result of the test
+
+    Replacement of Dict[str, Any]
+    """
+    log: Union[str, List[str]]
```

Comment:
Dangerous, can we stick to one type? If there's a single log, `List[str]` with just that single item would be easier for implementation.